concepts/dicts/about.md

```diff
@@ -1,55 +1,271 @@
+`Keys` can include `numbers`, `str`, `tuples` (of _immutable_ values), or `frozensets`, but must be hashable and unique across the dictionary.
+  `values` can be of any or multiple data type(s) or structures, including other dictionaries, built-in types, custom types, or even objects like functions.
 
-  Compared to searching for a value within a `list` or `array` (_without knowing the `index` position_), a `dict` uses significantly more space in memory, but has significantly more rapid retrieval.
- Dictionaries are especially useful in scenarios where the collection of items is large and must be accessed and/or updated frequently.
+`dict`s enable the retrieval of a `value` in (on average) constant O(1) time, given the `key`.
+Compared to searching for a value within a `list` or `array` (_without knowing the `index` position_), a dictionary uses significantly more memory, but has very rapid retrieval.
+Dictionaries are especially useful in scenarios where the collection of items is large and must be accessed and updated frequently.
 
-## Dictionary creation
 
-A simple `dict` can be declared using the literal form `{<key_1>: <value_1>, <key_2>: <value_2>}`:
+## Dictionary Construction
 
- ```python
+Dictionaries can be created in many ways.
+Among them are the [`fromkeys()`][fromkeys] method, [dictionary comprehensions][dict-comprehensions], merging two dictionaries via unpacking (`**`), merging via the `|` operator, or using a loop to iteratively add entries to a previously created empty `dict`.
 
+The two most straightforward methods of `dict` creation are using the `dict()` constructor or declaring a `dict` _literal_.
 
+### The `dict()` Class Constructor
 
+`dict()` can be used with any iterable of `key`, `value` pairs or with a series of `<name>=<value>` _arguments_:
 
+```python
+#Passing a list of key,value tuples.
+>>> wombat = dict([('name', 'Wombat'),('speed', 23),('land_animal', True)])
+{'name': 'Wombat', 'speed': 23, 'land_animal': True}
+
+
+#Using key=value arguments.
+>>> bear = dict(name="Black Bear", speed=40, land_animal=True)
+{'name': 'Black Bear', 'speed': 40, 'land_animal': True}
+```
+
+The [documentation on `dicts`][dicts-docs] outlines additional variations and options in constructor use.
+
+
+### Dictionary Literals
+
+A dictionary can also be directly entered as a _dictionary literal_, using curly brackets (`{}`) enclosing `key : value` pairs.
+Entries that are enclosed in the `{}` can also appear on separate lines:
+
+```python
+>>> whale = {"name": "Blue Whale", "speed": 35, "land_animal": False}
+{'name': 'Blue Whale', 'speed': 35, 'land_animal': False}
+
+>>> wombat = {'name': 'Wombat',
+              'speed': 23,
+              'land_animal': True,
+              'color': 'Brindle'}
+
+>>> wombat
+{'name': 'Wombat', 'speed': 23, 'land_animal': True, 'color': 'Brindle'}
+```
+
+Dictionaries can be arbitrarily nested:
+
+```python
+animal_collection = {
```

Comment:
Nice example! It might be to wide to be displayed correctly on the website?

Reply:
erg.  Yeah.  Big `dicts` are hard.  Do you know how wide the code window is?  I should format to that.

Reply:
Whelp, the code window is about 67 characters.  Tried to format it, but left some overlap.  It just doesn't read well if I do too much.